### DISCUSSION TITLE
Do both the python and C++ SDKs use the arrow::Buffer to hold payload under the hood?

### DISCUSSION THREAD

**Discussion by Zarca** [v6d-io/v6d#250] [Q&A] [answered]
Hi,
As the title refered,I'm thinking about :
1.Do both the python and C++ SDKs use the arrow::Buffer to hold payload under the hood?
2.Can one use the C++ to build Array type and Seal it into vineyard, and then resolve it using the Python SDK?

**Comment by sighingnow** [comment, marked as answer]
Hi @Zarca, sorry I missed this message!

> 1.Do both the python and C++ SDKs use the arrow::Buffer to hold payload under the hood?

Yes, actually, a blob is just a consecutive block of memory.

> 2.Can one use the C++ to build Array type and Seal it into vineyard, and then resolve it using the Python SDK?

Yes, definitiely you can, e.g., currently C++ and Python could share data types like vector, tensor (np.ndarray) and dataframe (pd.DataFrame).
